### PR TITLE
Add error handling for farnet location and GeoNames

### PR DESCRIPTION
**Pull request by bianchim**
MULTISITE-12727

### CODE REVIEW

File: tests/features/bootstrap/FeatureContext.php [hunk 1]
```diff
@@ -407,4 +409,51 @@ public function cleanFarnetErrorVariables() {
...
   }
 
+  /**
+   * Prepare the configuration for the use of GeoNames.
+   *
+   * @beforeScenario @GeoNames
+   */
+  public function geonamesBeforeScenario() {
+    $this->geonameUsername = variable_get('geonames_username', FALSE);
+    variable_set('geonames_username', 'bianchim');
```
Comment (gervasek) [inline tests/features/bootstrap/FeatureContext.php]: We cannot use a generic account instead of "bianchim" ?

Comment (gervasek) [inline tests/features/bootstrap/FeatureContext.php]: GeoNames watchdog should be deleted from database on the BeforScenario function

File: tests/features/farnet_gp_method.feature [hunk 2]
```diff
@@ -33,7 +33,7 @@ Feature: Factsheet GP Method content type
     And I should see the error message "Title field is required."
     And I should see the error message "You must select one or more groups for this content."
 
-  @api
+  @api @GeoNames
```
Comment (gervasek) [inline tests/features/farnet_gp_method.feature]: Is it really necessary to limit geonamesAfterScenario with the "@GeoNames" tag ? Are we sure that GeoName service will be never used on other Behat scenario ?

File: lib/modules/farnet_location/farnet_location.module [hunk 3]
```diff
@@ -562,3 +566,73 @@ function _farnet_location_get_field_city_select_dropdown($country_selected, $pos
...
+ * @return bool | array
+ *   Returns the result or FALSE if no result could be returned.
+ */
+function _farnet_location_get_geonames_result($url) {
+  $response = drupal_http_request($url);
+  $geolocs = drupal_json_decode($response->data);
+
+  // If an error occurred during JSON processing, log error and quit.
+  if (json_last_error() !== JSON_ERROR_NONE) {
+    watchdog('GeoNames', 'GeoNames return was not a valid json, response : @response', ['@response' => $response->data], WATCHDOG_ERROR);
```
Comment (gervasek) [inline lib/modules/farnet_location/farnet_location.module]: Add $url in watchdog will be nice

Comment (gervasek) [inline lib/modules/farnet_location/farnet_location.module]: Add ```$response->code``` on watchdog can be useful !

**Comment by bianchim** [comment]
- Remove the GeoNames behat tag
- Added info when API calls fail
- Changed geonames account

**Pull request merged**